llvm/lib/Target/X86/MCTargetDesc/X86ATTInstPrinter.cpp

```diff
@@ -205,7 +214,8 @@ bool X86ATTInstPrinter::printVecCompareInstr(const MCInst *MI,
               printwordmem(MI, CurOp--, OS);
             else
               printdwordmem(MI, CurOp--, OS);
-          } else if ((Desc.TSFlags & X86II::OpPrefixMask) == X86II::XD) {
+          } else if ((Desc.TSFlags & X86II::OpPrefixMask) == X86II::XD &&
```

Comment:
This check in X86IntelInstPrinter looks very similar, but isn't updated by this commit: https://github.com/llvm/llvm-project/blob/ede40da1f8c1e91601b985cd32ad785aa8806880/llvm/lib/Target/X86/MCTargetDesc/X86IntelInstPrinter.cpp#L207

We're getting an assertion there. Did you mean to also update the same check in X86IntelInstPrinter?

Reply:
Good catch. Will update. And the assert should be removed.

Reply:
addressed in https://github.com/llvm/llvm-project/pull/107529